com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/mqtt_streams.py

```diff
@@ -0,0 +1,208 @@
+ #      The password needed to access the encrypted trustStore file.
+ #      </td></tr>
+ # <tr><td>keyStore</td>
+ #      <td>Optional String. The pathname to a file containing the
+ #      MQTT client's public private key certificates.
+ #      If a relative path is specified, the path is relative to the
+ #      application directory. 
+ #      Required when an MQTT server is configured to use SSL client authentication.
+ #      </td></tr>
+ # <tr><td>keyStorePassword</td>
+ #      <td>Required String when {@code keyStore} is used.
+ #      The password needed to access the encrypted keyStore file.
+ #      </td></tr>
+ # <tr><td>receiveBufferSize</td>
+ #      <td>[subscribe] Optional Integer. The size, in number
+ #      of messages, of the subscriber's internal receive buffer.  Received
+ #      messages are added to the buffer prior to being converted to a
+ #      stream tuple. The receiver blocks when the buffer is full.
+ #      The default is 50.
+ #      </td></tr>
+ # <tr><td>retain</td>
+ #      <td>[publish] Optional Boolean. Indicates if messages should be
+ #      retained on the MQTT server.  Default is false.
+ #      </td></tr>
+ # <tr><td>defaultQOS</td>
+ #      <td>Optional Integer. The default
+ #      MQTT quality of service used for message handling.
+ #      The default is 0.
+ #      </td></tr>
+ # </table>
+ # 
+ # @see <a href="http://mqtt.org">http://mqtt.org</a>
+ # @see <a
+ #      href="http://ibmstreams.github.io/streamsx.messaging/">com.ibm.streamsx.messaging</a>
+
+
+    def __init__(self,topology,config) :
+        self.topology = topology
+        self.config = config.copy()
+        self.opCnt = 0
+        
+    def publish(self,src_op,topic=None) :
+        if topic is None:
+            raise TypeError("No topic provided to MqttStreams.publish")
+        parms = self.config.copy()
+        #parms['reconnectionBound'] = int(-1) (needs to be int vs long)
+        #parms['qos'] = int(0) (needs to be int vs long)
+        #del parms['messageQueueSize']
+        if topic is None :
+            parms['topicOutAttrName'] = "topic"
+        else :
+            parms['topic'] = topic
+        parms['dataAttributeName'] = "message"
+        if (++self.opCnt > 1) :
+            # each op requires its own clientID
+            clientId =  params['clientID']
+            if (clientId is None and len(clientId) > 0) :
+                params['clientID'] = opCnt+"-"+clientId
+        # convert src_op outputport schema from spl po to spl rstring type
+        forOp = self.topology.graph.addOperator("com.ibm.streamsx.topology.functional.python::PyFunctionFormat")
+        forOp.addInputPort(outputPort=src_op.oport)
+        forOport = forOp.addOutputPort(schema=schema.StreamSchema("tuple<rstring message>"))                
+        op = self.topology.graph.addOperator(kind="com.ibm.streamsx.messaging.mqtt::MQTTSink")
+        oport = op.addOutputPort(schema=schema.StreamSchema("tuple<rstring message>"))
```

Comment:
I think the MQTTSink operator requires an rstring attribute called message ie: tuple<rstring message>, in CommonSchema the attribute is called "string" ie:  tuple<rstring string>.  Using CommonSchema.String, I get the following error:
23 May 2016 16:32:31.364 [6806] ERROR #splapptrc,J[120],P[120],test_TopologyMQTTpublish_OP2 M[?:com.ibm.streams.operator.internal.compile.OperatorContextCheckerImpl.errorRequiredAttribute:-1]  - Required attribute message is missing from schema tuple<rstring string>.